tests/integration_tests/build/test_coverage.py

```diff
@@ -29,109 +23,92 @@
-        "--target {} --output {} -- "
-        "--exclude-pattern={} "
-        "--exclude-region={} --verify"
-    ).format(
-        host.get_rustflags(),
-        os.path.join(test_fc_session_root_path, CARGO_KCOV_REL_PATH),
-        target,
-        test_session_tmp_path,
-        exclude_pattern,
-        exclude_region,
-    )
-    # We remove the seccompiler custom build directory, created by the
-    # vmm-level `build.rs`.
-    # If we don't delete it before and after running the kcov command, we will
-    # run into linker errors.
-    shutil.rmtree(SECCOMPILER_BUILD_DIR, ignore_errors=True)
-    # By default, `cargo kcov` passes `--exclude-pattern=$CARGO_HOME --verify`
-    # to kcov. To pass others arguments, we need to include the defaults.
-    utils.run_cmd(cmd)
-
-    shutil.rmtree(SECCOMPILER_BUILD_DIR)
-
-    coverage_file = os.path.join(test_session_tmp_path, KCOV_COVERAGE_FILE)
-    with open(coverage_file, encoding="utf-8") as cov_output:
-        contents = cov_output.read()
-        covered_lines = int(re.findall(KCOV_COVERED_LINES_REGEX, contents)[0])
-        total_lines = int(re.findall(KCOV_TOTAL_LINES_REGEX, contents)[0])
-        coverage = covered_lines / total_lines * 100
-    print("Number of executable lines: {}".format(total_lines))
-    print("Number of covered lines: {}".format(covered_lines))
-    print("Thus, coverage is: {:.2f}%".format(coverage))
-
-    coverage_low_msg = (
-        "Current code coverage ({:.2f}%) is >{:.2f}% below the target ({}%).".format(
-            coverage, COVERAGE_MAX_DELTA, coverage_target_pct
-        )
+    # Get coverage target.
+    processor_model = [item for item in COVERAGE_DICT if item in PROC_MODEL]
+    assert len(processor_model) == 1, "Could not get processor model!"
+    coverage_target = COVERAGE_DICT[processor_model[0]]
+
+    # Re-direct to repository root.
+    monkeypatch.chdir("..")
+
+    # Generate test profiles.
+    utils.run_cmd(
+        f'\
+        env RUSTFLAGS="-Cinstrument-coverage" \
+        LLVM_PROFILE_FILE="coverage-%p-%m.profraw" \
+        cargo test --all --target={TARGET} -- --test-threads=1 \
+    '
     )
 
-    assert coverage >= coverage_target_pct - COVERAGE_MAX_DELTA, coverage_low_msg
-
-    # Get the name of the variable that needs updating.
-    namespace = globals()
-    cov_target_name = [name for name in namespace if namespace[name] is COVERAGE_DICT][
-        0
-    ]
-
-    coverage_high_msg = (
-        "Current code coverage ({:.2f}%) is >{:.2f}% above the target ({}%).\n"
-        "Please update the value of {}.".format(
-            coverage, COVERAGE_MAX_DELTA, coverage_target_pct, cov_target_name
-        )
+    # Generate coverage report.
+    utils.run_cmd(
+        f'\
+        cargo install --version {GRCOV_VERSION} grcov \
+        && grcov . \
+            -s . \
+            --binary-path ./build/cargo_target/{TARGET}/debug/ \
+            --excl-start "mod tests" \
+            --ignore "build/*" \
```

Comment:
Does this exclude list match what we had before? The old list is 
```
      "${CARGO_HOME:-$HOME/.cargo/},"
        "build/,"
        "tests/,"
        "usr/lib/gcc,"
        "lib/x86_64-linux-gnu/,"
        "test_utils.rs,"
        # The following files/directories are auto-generated
        "bootparam.rs,"
        "elf.rs,"
        "mpspec.rs,"
        "msr_index.rs,"
        "bindings.rs,"
        "_gen"
```

Reply:
Ignoring `build/*` relates more to fixing awkward behavior resulting from our non-default build directory. Otherwise `grcov` only evaluates coverage within `src`.

We shouldn't ignore auto-generated rust files. I would argue this was a mistake originally.

Reply:
Oh I see. I don't have strong opinions on the auto generated files thing, but bindgen doesn't use `#[cfg(test)]` to gatekeep its tests, so the reports contain a lot of autogenerated tests at the moment. Maybe including the generated files in the coverage report would be a discussion for a separate PR?

Independently of that, the `test_utils.rs` files and `tests/` modules used to be excluded from reports, and are not anymore.

Reply:
I agree `tests/` should be ignored, I will add this.

I would argue the `test_utils.rs` files should be included. The `test_utils.rs` not `cfg(test)` so we can only consider them to be production code.

Reply:
`test_utils` are not `#[cfg(test)]` because they are meant to be usable from non-unittests (e.g. when cargo compiled integration tests, it does not pass `--cfg test` to the library under test). It would be a logic error if any of their code was actually executed in a production context. 

Let's get the second reviewers opinion on this

Reply:
This I would argue is a design flaw, `test_utils.rs` should not exist as they presently do. But I do not think we can ignore coverage on code which is compiled when building a release binary.

Reply:
I agree that long term these `test_utils.rs` files should probablity live in their own crate which is not linked into a release binary (although currently their code is removed by link time dead code elimination, as they are not used outside of testing code).

Reply:
> (although currently their code is removed by link time dead code elimination, as they are not used outside of testing code).

I'm not sure this is guaranteed.

Reply:
> This I would argue is a design flaw, the `test_utils.rs` files should not exist as they presently do. But I do not think we can ignore coverage on code which is compiled when building a release binary.

I would argue to not include changes in this PR that are not in parity with what used to be before (I.e let's target only to update coverage tool).
If we feel there are other changes that need to be made, we want to put them in a separate commit/PR and properly motivate them.

Reply:
Added `test_utils.rs` to ignore list.